File: CV_Tasks/Task_1/python/main.py
import cv2

def main():
    image = cv2.imread("../resources/test.png")

    if image is None:
        print("Error: couldn't open image")
        exit()

    image = cv2.resize(image, (500, 500))

    image = cv2.flip(image, -1)

    image = image[0:400, 0:400]

    height, width = image.shape[:2]

    rotate_matrix_45 = cv2.getRotationMatrix2D((width / 2, height / 2), 45.0, 1.0)
    image = cv2.warpAffine(image, rotate_matrix_45, (width, height))

    cv2.imshow("changed image", image)
    cv2.imwrite("new.png", image)

    cv2.waitKey(0)
    cv2.destroyAllWindows()


if __name__ == "__main__":
    main()

File: CV_Tasks/Task_1/c++/main.cpp
#include <opencv2/opencv.hpp>

int main() {
	cv::Mat image = cv::imread("../../resources/test.png");

	if (image.empty()) {
		std::cerr << "Couldn't open the image" << std::endl;
		return 1;
	}

	cv::resize(image, image, cv::Size(1000, 1000));

	cv::flip(image, image, -1);

	cv::Rect crop(0, 0, 900, 900);
	image = image(crop);

	cv::Point2f center(image.cols / 2, image.rows / 2);
	cv::Mat rotate_matrix_45 = cv::getRotationMatrix2D(center, 45.0, 1.0);
	cv::warpAffine(image, image, rotate_matrix_45, image.size());

	cv::imshow("changed photo", image);
	cv::imwrite("new.png", image);

	cv::waitKey(0);
	cv::destroyAllWindows();
}
